client/blocks/importer/wordpress/import-everything/import-users/index.tsx

```diff
@@ -0,0 +1,197 @@
+import { useSendInvites } from '@automattic/data-stores';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { NextButton, Title, SubTitle } from '@automattic/onboarding';
+import { Button } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import ExternalLink from 'calypso/components/external-link';
+import InfiniteList from 'calypso/components/infinite-list';
+import useExternalContributorsQuery from 'calypso/data/external-contributors/use-external-contributors';
+import useP2GuestsQuery from 'calypso/data/p2/use-p2-guests-query';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import ImportedUserItem from './imported-user-item';
+import { getRole } from './utils';
+import type { UsersQuery, Member } from '@automattic/data-stores';
+
+import './style.scss';
+
+interface Props {
+	site: {
+		ID: number;
+	};
+	onSubmit: () => void;
+}
+
+const ImportUsers = ( { site, onSubmit }: Props ) => {
+	const defaultTeamFetchOptions = { include_viewers: true };
+	const userId = useSelector( getCurrentUserId );
+	const translate = useTranslate();
+	const usersQuery = useUsersQuery( site?.ID, defaultTeamFetchOptions ) as unknown as UsersQuery;
+	const { data: externalContributors } = useExternalContributorsQuery( site?.ID );
+	const { data: p2Guests } = useP2GuestsQuery( site?.ID );
+	const { data: usersData, fetchNextPage, isFetchingNextPage, hasNextPage } = usersQuery;
+	const { isPending: isSubmittingInvites, mutateAsync: sendInvites } = useSendInvites( site?.ID );
+
+	const users = usersData?.users?.map( ( user ) => ( { user, checked: true } ) ) || [];
+	const [ usersList, setUsersList ] = useState( users );
+	const [ checkedUsersNumber, setCheckedUsersNumber ] = useState( usersList?.length || 0 );
+
+	const handleSubmit = async () => {
+		const selectedUsers = usersList
+			.filter( ( user ) => user.checked )
+			.map( ( userItem ) => ( {
+				email_or_username:
+					typeof userItem.user?.email === 'string' ? userItem.user?.email : userItem.user?.login,
+				role: getRole( userItem.user ),
+			} ) );
+
+		if ( selectedUsers.length === 0 ) {
+			return;
+		}
+
+		recordTracksEvent( 'calypso_site_importer_import_users_submit_invite', {
+			number_of_invites: selectedUsers.length,
+		} );
+
+		await sendInvites( selectedUsers )
+			.then( ( response ) => {
```

Comment:
Maybe a try catch would be more readable here? Since it uses await.
```
try {
    const result = await sendInvites(selectedUsers);
	... rest of the logic ...
} catch (error) {
    setUserInviteError('generic_error');
}
```

Reply:
Updated to include try catch.

@renancarvalho / @escapemanuele / @crisbusquets - We discussed error messages/states and Cris proposed continuing the flow regardless of an invite API failure/error for a cleaner user experience (wont display an error as users are migrating to your, which doesn't look so great from a user's perspective )

Maybe we should redirect users to `wp-admin/users`  at the end of the flow if there any errors and provide a message notifying them that can invite users to their blog (or simply redirect)?

Reply:
I think it makes sense 👍  Go for it @agrullon95